cfme/tests/services/test_service_rbac.py

```diff
@@ -0,0 +1,121 @@
+
+
+def new_role(appliance, product_features):
+    collection = appliance.collections.roles
+    return collection.create(name='role_{}'.format(fauxfactory.gen_alphanumeric()),
+                             vm_restriction=None, product_features=product_features)
+
+
+def new_group(appliance, role):
+    collection = appliance.collections.groups
+    return collection.create(description='group_{}'.format(fauxfactory.gen_alphanumeric()),
+                             role=role, tenant="My Company")
+
+
+def new_user(appliance, group, credential):
+    collection = appliance.collections.users
+    return collection.create(name='user_{}'.format(fauxfactory.gen_alphanumeric()),
+                             credential=credential,
+                             email='xyz@redhat.com',
+                             groups=group,
+                             cost_center='Workload',
+                             value_assign='Database')
+
+
+@pytest.yield_fixture(scope='module')
+def role_user_group(appliance, new_credential):
+    role = new_role(appliance=appliance, product_features=[(['Everything'], False),
+                            (['Everything', 'Access Rules for all Virtual Machines'], True)])
+    group = new_group(appliance=appliance, role=role.name)
+    user = new_user(appliance=appliance, group=group, credential=new_credential)
+    yield role, user
+
+
+def test_service_rbac_no_permission(appliance, role_user_group):
+    """ Test service rbac without user permission """
+    role, user = role_user_group
+    error_message = ("The user's role is not authorized for any access, "
+                     "please contact the administrator!")
+    with pytest.raises(Exception, match=error_message):
+        appliance.server.login(user)
+
+
+@pytest.mark.parametrize('permission', CATALOG_MODULES)
+def test_service_rbac_catalog_explorer(appliance, role_user_group, catalog,
+                                       catalog_item, permission):
+    """ Test service rbac with only catalog explorer module permissions"""
+    role, user = role_user_group
+    if permission == CATALOG_MODULES[3]:
+        product_features = [
+            (['Everything'], True), (['Everything'], False),
+            (['Everything', 'Services', 'Requests'], True),
+            (['Everything', 'Automation', 'Automate', 'Customization'], True)
+        ]
+        product_features.extend([(['Everything', 'Services', 'Catalogs Explorer', k], True)
+                                 for k in ['Catalog Items', 'Service Catalogs', 'Catalogs']])
+        role.update({'product_features': product_features})
+    else:
+        product_features = [(['Everything'], True), (['Everything'], False)]
+        product_features.extend([(['Everything', 'Services', 'Catalogs Explorer', k], True)
+                                 for k in [permission]])
+        role.update({'product_features': product_features})
+
+    with user:
+        if permission == CATALOG_MODULES[0]:
```

Comment:
**REQUIRED:** I didn't pick up on this before, but this is potentially going to yield very unexpected results if that list is altered. The strings would be a fine alternative here. I would even posit that it would make more sense here to make 4 separate tests.